posthog-node/test/feature-flags.spec.ts

```diff
@@ -2051,81 +1987,74 @@ describe('match properties', () => {
     expect(matchProperty(property_d, { key: '2022-04-05 11:34:13 +00:00' })).toBe(false)
   })
 
-  it('with relative date operators', () => {
-    jest.setSystemTime(new Date('2022-05-01'))
-
-    const property_a = { key: 'key', value: '-6h', operator: 'is_date_before' }
-    expect(matchProperty(property_a, { key: '2022-03-01' })).toBe(true)
-    expect(matchProperty(property_a, { key: '2022-04-30' })).toBe(true)
-
+  it.each([
+    ['is_date_before', '-6h', '2022-03-01', false],
+    ['is_date_before', '-6h', '2022-04-30', true],
     // :TRICKY: MonthIndex is 0 indexed, so 3 is actually the 4th month, April.
-    expect(matchProperty(property_a, { key: new Date(Date.UTC(2022, 3, 30, 1, 2, 3)) })).toBe(true)
+    ['is_date_before', '-6h', new Date(Date.UTC(2022, 3, 30, 1, 2, 3)), true],
```

Comment:
I had to do some refactoring here as the tests were really hard to debug and only seemed to start failing in this PR (still figuring out why)

Reply:
Not sure if anything has changed here 🤔 , very happy to keep this refactor, but could you please first revert & check tests still pass, and then restore?